packages/template-retail-react-app/app/components/social-login/index.jsx

```diff
@@ -19,21 +20,31 @@ import {AppleIcon, GoogleIcon} from '@salesforce/retail-react-app/app/components
  * @returns
  */
 const SocialLogin = ({idps}) => {
+    const intl = useIntl()
     const IDP_CONFIG = {
         apple: {
             icon: AppleIcon,
-            message: 'Apple'
+            message: intl.formatMessage({
+                id: 'login_form.button.apple',
+                defaultMessage: 'Apple'
+            })
         },
         google: {
             icon: GoogleIcon,
-            message: 'Google'
+            message: intl.formatMessage({
+                id: 'login_form.button.google',
+                defaultMessage: 'Google'
+            })
         }
     }
 
     return (
         idps && (
             <Stack spacing={4}>
                 {idps.map((name) => {
+                    if (!(name in IDP_CONFIG)) {
+                        logger.error('IDP "'+ name + '" is missing from IDP_CONFIG. Valid IDPs are ['+ Object.keys(IDP_CONFIG).join(', ') + '].')
+                    }
```

Comment:
Why don't we return null and stop the code earlier from running any check below if the config is missing?